File: apps/backendts/src/services.ts
import {
  streamEvent,
  openaiChatCompletionsStream,
  assistantGetTools,
  todosUpdate,
  toolHnSearch,
  toolUpdateTodos,
} from "./functions";
import { client } from "./client";
import { openaiService } from "@restackio/integrations-openai";

export async function services() {
  const workflowsPath = require.resolve("./workflows");
  try {
    await Promise.all([
      // Start service with current workflows and functions
      client.startService({
        workflowsPath,
        functions: {
          streamEvent,
          openaiChatCompletionsStream,
          assistantGetTools,
          todosUpdate,
          toolHnSearch,
          toolUpdateTodos,
        },
      }),
      openaiService({ client }),
    ]);

    console.log("Services running successfully.");
  } catch (e) {
    console.error("Failed to run services", e);
  }
}

services().catch((err) => {
  console.error("Error running services:", err);
});

File: apps/backendpy/src/services.py
import asyncio
from src.functions.openai.chat import openai_todos
from src.functions.hn.search import tool_hn_search
from src.client import client
from src.workflows.automated import AutomatedWorkflow

async def main():

    await client.start_service({
        "workflows": [AutomatedWorkflow],
        "functions": [openai_todos, tool_hn_search]
    })

def run_services():
    asyncio.run(main())

if __name__ == "__main__":
    run_services()